openlibrary/solr/query_utils.py

```diff
@@ -279,3 +278,62 @@ def luqum_replace_field(query, replacer: Callable[[str], str]) -> str:
         if isinstance(sf, SearchField):
             sf.name = replacer(sf.name)
     return str(query)
+
+
+def luqum_field_traverse(item: Item, _parents: list[Item] | None = None):
+    """
+    Unlike the normal field traversal, this function will not go deeper once it hits a search field.
+    THis is necessary, in order to properly avoid fuzzying out undesired fields in luqum_make_fuzzy.
+
+    Nuch like its sister function, it does not make any guarantees about what will happen if you
+    modify the tree while traversing it 😅 But we do it anyways.
+
+    :param item: Node to traverse
+    :param _parents: Internal parameter for tracking parents
+    """
+    parents = _parents or []
+    yield item, parents
+    new_parents = [*parents, item]
+    if not isinstance(item, SearchField):
+        for child in item.children:
+            yield from luqum_field_traverse(child, new_parents)
+
+
+def luqum_make_fuzzy(tree: Item, search_fields: set[str] | None = None):
+    """:param query: Passed in the form of a luqum tree
+    :param search_fields: A list of search fields to be fuzzied out."""
```

Comment:
Let's add some more tests in this doc string testing some edge cases eg

- parentheses
- quotations
- greedy fields, eg `lord of the rings author:j r r tolkein`

Reply:
It looks like the tests were already largely included; I added one additional one for the parentheses, but all of the other situations seemed to be covered.

Reply:
Done